mongodb/src/main/scala/monix.connect.mongodb/MongoSink.scala

```diff
@@ -0,0 +1,300 @@
+    implicit
+    collection: MongoCollection[Doc]): Consumer[Seq[Doc], Unit] = {
+    insertMany(Some(insertManyOptions), retries, Option.empty[FiniteDuration])(collection)
+  }
+
+  def insertMany[Doc](retries: Int, timeout: FiniteDuration)(
+    implicit
+    collection: MongoCollection[Doc]): Consumer[Seq[Doc], Unit] = {
+    insertMany(Option.empty[InsertManyOptions], retries, Option.empty[FiniteDuration])(collection)
+  }
+
+  private def updateOne[Doc](updateOptions: Option[UpdateOptions], retries: Int, timeout: Option[FiniteDuration])(
+    collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
+    val updateOp = (t: Tuple2[Bson, Bson]) => {
+      updateOptions match {
+        case Some(options) => collection.updateOne(t._1, t._2, options)
+        case None => collection.updateOne(t._1, t._2)
+      }
+    }
+    new MongoSink[(Bson, Bson)](updateOp, retries, timeout)
+  }
+
+  def updateOne[Doc](collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
+    updateOne(Option.empty[UpdateOptions], 1, Option.empty[FiniteDuration])(collection)
+  }
+
+  def updateOne[Doc](updateOptions: UpdateOptions)(
+    implicit
+    collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
+    updateOne(Some(updateOptions), 1, Option.empty[FiniteDuration])(collection)
+  }
+
+  def updateOne[Doc](updateOptions: UpdateOptions, retries: Int)(
+    implicit
+    collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
+    updateOne(Some(updateOptions), retries, Option.empty[FiniteDuration])(collection)
+  }
+
+  def updateOne[Doc](updateOptions: UpdateOptions, retries: Int, timeout: FiniteDuration)(
+    implicit
+    collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
+    updateOne(Some(updateOptions), retries, Option.empty[FiniteDuration])(collection)
+  }
+
+  def updateOne[Doc](retries: Int, timeout: FiniteDuration)(
+    implicit
+    collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
+    updateOne(Option.empty[UpdateOptions], retries, Option.empty[FiniteDuration])(collection)
+  }
+
+  private def updateMany[Doc](updateOptions: Option[UpdateOptions], retries: Int, timeout: Option[FiniteDuration])(
+    implicit
+    collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
+    val updateOp = (t: Tuple2[Bson, Bson]) => {
+      updateOptions match {
+        case Some(options) => collection.updateMany(t._1, t._2, options)
+        case None => collection.updateMany(t._1, t._2)
+      }
+    }
+    new MongoSink[(Bson, Bson)](updateOp, retries, timeout)
+  }
+
+  def updateMany[Doc](collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
+    updateMany(Option.empty[UpdateOptions], 1, Option.empty[FiniteDuration])(collection)
+  }
+
+  def updateMany[Doc](updateOptions: UpdateOptions)(
+    implicit
+    collection: MongoCollection[Doc]): Consumer[(Bson, Bson), Unit] = {
```

Comment:
@alexandru, @Avasil what do you think on defining collection as implicit? I like to provide the api as an object and minimise having to pass the collection explicitly every time but on the other hand if the user have to use different collections it can be a bit verbose. 
it would also be possible to provide an alternative method that does not use implicits but I think is better to stick to one to avoid confusions.